web/src/logfilters.ts

```diff
@@ -61,9 +86,23 @@ export function filterSetFromLocation(l: Location): FilterSet {
       break
   }
 
+  const sourceTerm = params.get("term")
+  if (sourceTerm) {
+    filters.term = {
+      sourceTerm,
+      filterTerm: parseFilterTerm(sourceTerm),
+      invalid: false,
```

Comment:
Yeah, I was thinking that if a user enters a regexp that we can't parse, we could set `invalid` to `true`. (This pr is a little funny because it's only parsing strings and we technically won't hit an error state just yet.) The design doesn't include an error state, but at the minimum, I wanted to pass that error state to an ARIA property on the term field.

I like the state pattern you outlined and will update my code. 🔧